src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java

```diff
@@ -62,13 +63,32 @@
     private final String apiUri;
     private String pattern = ".*";
 
+    private @CheckForNull String includes;
+    private @CheckForNull String excludes;
+
     @DataBoundConstructor public GitHubSCMNavigator(String apiUri, String repoOwner, String scanCredentialsId, String checkoutCredentialsId) {
         this.repoOwner = repoOwner;
         this.scanCredentialsId = Util.fixEmpty(scanCredentialsId);
         this.checkoutCredentialsId = checkoutCredentialsId;
         this.apiUri = Util.fixEmpty(apiUri);
     }
 
+    public @Nonnull String getIncludes() {
+        return includes != null ? includes : DescriptorImpl.defaultIncludes;
+    }
+
+    @DataBoundSetter public void setIncludes(@Nonnull String includes) {
+        this.includes = includes.equals(DescriptorImpl.defaultIncludes) ? null : includes;
+    }
+
+    public @Nonnull String getExcludes() {
```

Comment:
A contributor recently arrived. He never used this annotation but he looks the file and figures out three different *configurations* or *usages* of this annotation. **What do you think it can be his first thought?** *Oh, in this file have contributed three different coders with different coding styles and criteria* or *Is it possible that this annotation has different behavior according his location?*

In my opinion, are small details can do the life more easier. Isn't that so?